<br><br>
# <center>STATISTICAL METHODS IN FINANCE</center>
# <center>HW6</center>
<center>Fan Yang</center>
<center>UNI: fy2232</center>
<center>03/09/2018</center>

# Problem 1

***Suppose that the daily return on a stock is normally distributed with mean 0.002 and standard deviation 0.016. Suppose that you invest on this stock. What is the probability that your loss in one day is more than one standard deviation above the mean loss?***

the **loss** on this stock is normally distributed with mean **-0.002** and standard deviation 0.016

In [1]:
1-pnorm(0.016-0.002,mean = -0.002,sd = 0.016)

[1] 0.1586553

$$Pr(L>\sigma+\mu)=1-Pr(L\leq0.014)=0.158655$$

# Problem 2

***(a)
Suppose that the daily return on a stock is normally distributed with mean 0.002 and standard deviation 0.016. Suppose that you invest \$1,000,000 on this stock. What is the VaR(0.1) with T (time unit) equal to one day?***

$z_{1-\alpha}=1.28155$<br>
the value at risk of the portfolio at $\alpha = 0.1$ is
\begin{split}
\text{VaR}(0.1)&=1000000\times(\sigma z_{1-\alpha}-\mu)\\
&=1000000\times(0.016\times z_{1-\alpha}-0.002)\\
&=1000000\times(0.016\times 1.28155-0.002)\\
&=1000000\times0.0185048\\
&=18504.8\\
\end{split}

***(b) Suppose that the yearly return of a stock follows a normal distribution with mean 0.1 and standard deviation 0.2. Suppose that you buy \$1,000 worth of this stock. What is the VaR(0.05) with T equal to one year?***

$z_{1-\alpha}=1.64485$<br>
the value at risk of the portfolio at $\alpha = 0.05$ is
\begin{split}
\text{VaR}(0.05)&=1000\times(\sigma z_{1-\alpha}-\mu)\\
&=1000\times(0.2\times z_{1-\alpha}-0.1)\\
&=1000\times(0.2\times 1.64485-0.1)\\
&=1000\times0.22897\\
&=228.97\\
\end{split}

# Problem 3

***(a)
Suppose that the daily return of a stock, R, satisfies that $\frac{R−\mu}{\sigma} \sim t(2)$, where $t(2)$ stands for a central t-distribution with 2 degrees of freedom. Suppose that $\mu = 0.002$ and $\sigma = 0.016$. Suppose that you buy \$1000 worth of this stock. What is the VaR(0.1) with T equal to one day?***

$t_{1-\alpha}(2)=1.885618$<br>
the value at risk of the portfolio at $\alpha = 0.1$ is
\begin{split}
\text{VaR}(0.1)&=1000\times(\sigma t_{1-\alpha}(2)-\mu)\\
&=1000\times(0.016\times t_{1-\alpha}(2)-0.002)\\
&=1000\times(0.016\times 1.885618-0.002)\\
&=1000\times0.028169889\\
&=28.169889\\
\end{split}

***(b)
Suppose that the daily return of a stock, R, satisfies that $\frac{R−\mu}{\sigma} \sim t(5)$. Suppose that $\mu = 0.002$ and $\sigma = 0.016$. Suppose that you buy \$1000 worth of this stock. What is the VaR(0.1) with T equal to one day?***

$t_{1-\alpha}(5)=1.475884$<br>
the value at risk of the portfolio at $\alpha = 0.1$ is
\begin{split}
\text{VaR}(0.1)&=1000\times(\sigma t_{1-\alpha}(5)-\mu)\\
&=1000\times(0.016\times t_{1-\alpha}(5)-0.002)\\
&=1000\times(0.016\times 1.475884-0.002)\\
&=1000\times0.02161414\\
&=21.61414478\\
\end{split}

# Problem 4

***Suppose that you invest \$500 on stock A and $1000 on stock B. Suppose that the yearly return of stock A is normal with mean 0.01 and standard deviation 0.05 and the yearly return of stock B is normal with mean 0.005 and standard deviation 0.01. Further assume that they are jointly normal. Calculate VaR(0.05) when***
>***(a)*** they are independent;<br> 
***(b)*** their correlation is 0.3;<br>
***(c)*** their correlation is -0.3.

We invest \$500 on A and \$1000 on B, so the weight is $w_A=\frac{1}{3}$ and $w_B=\frac{2}{3}$.<br>
$z_{1-\alpha}=1.64485$

## (a)
\begin{split}
\mu_{A+B}&=w_A\mu_A+w_
B\mu_B=\frac{1}{3}\times0.01+\frac{2}{3}\times0.005\\
&=\frac{1}{150}\\
&=0.0066666667\\
\sigma_{A+B}&=\sqrt{(w_A\sigma_A)^2+(w_
B\sigma_B)^2}=\sqrt{(\frac{1}{3}\times0.05)^2+(\frac{2}{3}\times0.01)^2}\\
&=0.017950549\\\\
\text{VaR}(0.05)&=(500+1000)\times(\sigma z_{1-\alpha}-\mu)\\
&=1500\times(0.01795\times Z_{1-\alpha}-0.00667)\\
&=1500\times(0.01795\times 1.64485-0.00667)\\
&=1500\times0.022859\\
&=34.289\\
\end{split}

## (b)
\begin{split}
\mu_{A+B}
&=0.0066666667\\
\sigma_{A+B}&=\sqrt{(w_A\sigma_A)^2+(w_
B\sigma_B)^2+2\rho w_A\sigma_A w_B\sigma_B}\\
&=\sqrt{(\frac{1}{3}\times0.05)^2+(\frac{2}{3}\times0.01)^2+
2\times0.3\times\frac{1}{3}\times0.05\times\frac{2}{3}\times0.01}\\
&=0.0197202659\\\\
\text{VaR}(0.05)&=(500+1000)\times(\sigma z_{1-\alpha}-\mu)\\
&=1500\times(0.01972\times Z_{1-\alpha}-0.0066667)\\
&=1500\times(0.01972\times 1.64485-0.0066667)\\
&=1500\times0.02577\\
&=38.655\\
\end{split}

## (c)
\begin{split}
\mu_{A+B}
&=0.0066666667\\
\sigma_{A+B}&=\sqrt{(w_A\sigma_A)^2+(w_
B\sigma_B)^2+2\rho w_A\sigma_A w_B\sigma_B}\\
&=\sqrt{(\frac{1}{3}\times0.05)^2+(\frac{2}{3}\times0.01)^2-
2\times0.3\times\frac{1}{3}\times0.05\times\frac{2}{3}\times0.01}\\
&=0.015986\\\\
\text{VaR}(0.05)&=(500+1000)\times(\sigma z_{1-\alpha}-\mu)\\
&=1500\times(0.015986\times Z_{1-\alpha}-0.0066667)\\
&=1500\times(0.015986\times 1.64485-0.0066667)\\
&=1500\times0.0196279\\
&=29.44185\\
\end{split}

# Problem 5

***Suppose that L is the loss over 1 year which follows a distribution with density$$
f(x)=\frac{1}{Z}\frac{|x+1|}{(x^2+1)^2},~x\in \mathbb{R},
$$
where Z is the normalizing constant.***

>***(a)*** Compute Z<br>
***(b)*** Calculate VaR(0.05).

## (a)

\begin{split}
Z=&\int_{-\infty}^{\infty}\frac{|x+1|}{(x^2+1)^2}dx\\
=&\int_{-\infty}^{-1}\frac{-(x+1)}{(x^2+1)^2}dx +
\int_{-1}^{\infty}\frac{x+1}{(x^2+1)^2}dx\\
=&-\int_{-\infty}^{-1}\frac{x+1}{(x^2+1)^2}dx +
\int_{-1}^{\infty}\frac{x+1}{(x^2+1)^2}dx\\
=&-\frac{1}{2}\left[\frac{x-1}{x^2+1}+\tan^{-1}(x)\right]\Bigg|_{-\infty}^{-1}+\frac{1}{2}\left[\frac{x-1}{x^2+1}+\tan^{-1}(x)\right]\Bigg|_{-1}^{\infty}\\
=&-\frac{1}{2}\left[(\frac{-2}{2}-\frac{\pi}{4})-(-\frac{\pi}{2})\right]+
\frac{1}{2}\left[(\frac{\pi}{2})-(\frac{-2}{2}-\frac{\pi}{4})\right]\\
=&-\frac{1}{2}\left[-1+\frac{\pi}{4}\right]+
\frac{1}{2}\left[\frac{3\pi}{4}+1\right]\\
=&\frac{1}{2}\left(1-\frac{\pi}{4}+\frac{3\pi}{4}+1\right]\\
=&\frac{1}{2}\left(2+\frac{\pi}{2}\right)\\
=&1+\frac{\pi}{4}\\
\text{therefore, }Z=&1+\frac{\pi}{4}\\
\end{split}<br>
(In the process, $\tan^{-1}(x)$ is the same as $\text{arc}\tan(x)$. )

## (b)

Set $1-\alpha$ quantile as $q$, and it is obvious that $q>-1$, then<br><br>
\begin{split}
0.95=&\int_{-\infty}^{q}\frac{1}{Z}\frac{|x+1|}{(x^2+1)^2}dx\\
=&\int_{-\infty}^{-1}\frac{1}{Z}\frac{-(x+1)}{(x^2+1)^2}dx +
\int_{-1}^{q}\frac{1}{Z}\frac{x+1}{(x^2+1)^2}dx\\
=&-\frac{1}{Z}\int_{-\infty}^{-1}\frac{x+1}{(x^2+1)^2}dx +
\frac{1}{Z}\int_{-1}^{q}\frac{x+1}{(x^2+1)^2}dx\\
=&-\frac{1}{2Z}\left[\frac{x-1}{x^2+1}+\tan^{-1}(x)\right]\Bigg|_{-\infty}^{-1}+\frac{1}{2Z}\left[\frac{x-1}{x^2+1}+\tan^{-1}(x)\right]\Bigg|_{-1}^{q}\\
=&-\frac{1}{2Z}\left[(\frac{-2}{2}-\frac{\pi}{4})-(-\frac{\pi}{2})\right]+
\frac{1}{2Z}\left[(\frac{q-1}{q^2+1}+\tan^{-1}(q)-(\frac{-2}{2}-\frac{\pi}{4})\right]\\
=&-\frac{1}{2Z}\left[-1+\frac{\pi}{4}\right]+
\frac{1}{2Z}\left[(\frac{q-1}{q^2+1}+\tan^{-1}(q)+1+\frac{\pi}{4})\right]\\
=&\frac{1}{2Z}\left(1-\frac{\pi}{4}+\frac{q-1}{q^2+1}+\tan^{-1}(q)+1+\frac{\pi}{4})\right)\\
=&\frac{1}{2Z}\left(2+\frac{q-1}{q^2+1}+\tan^{-1}(q)\right)\\
\text{where }Z=&1+\frac{\pi}{4}.\\
\end{split}
Now our goal is to caculate $p$ which makes the above equation hold.

In [2]:
equa <- function(q){
    Z = 1 + pi/4
    result = 2 + (q-1)/(q^2+1) + atan(q)
    return(1/2/Z*result - 0.95)
}
uniroot(equa,c(2, 3))$root

[1] 2.463719

From the above result we conclude that $q=2.463719$ is the $95\%$ quantile of $f(x)$.<br>
$$P(L>VaR(0.05))=0.05\\
P(L<VaR(0.05))=0.95$$
Therefore, $VaR(0.95)=2.463719$.

# Problem 6

***(a)
Suppose that the loss of an asset L, is normal with mean $\mu$ and variance $\sigma^2$. Show that the expected shortfall is given by $ES(\alpha) = \mu + \sigma\phi(z_{\alpha})/\alpha$, where $z_{\alpha}$ is the $(1 − \alpha)$-th quantile of the standard normal distribution and $\phi(\cdot)$ is the density function of N(0,1).***

$z_{\alpha}$ is the $(1 − \alpha)$-th quantile of the standard normal distribution and L is the loss of an asset, so
\begin{split}
VaR(s)&=\sigma z_{s}+\mu\\
ES(\alpha)&=-\frac{1}{\alpha}\int_{-\infty}^{VaR(\alpha)}
x \frac{1}{\sqrt{2\pi} \sigma}e^{-\frac{(x-\mu)^2}{2\sigma^2}}dx\\
\end{split}
Since $\frac{x-\mu}{\sigma}\sim N(0,1)$, let $x=s\sigma+\mu$,then
\begin{split}
ES(1-\alpha)&=-\frac{1}{\alpha}\int_{-\infty}^{VaR(\alpha)}
x \frac{1}{\sqrt{2\pi} \sigma}e^{-\frac{(x-\mu)^2}{2\sigma^2}}~dx\\
&=-\frac{1}{\alpha}\int_{-\infty}^{\frac{VaR(\alpha)-\mu}{\sigma}}
\frac{s\sigma+\mu}{\sqrt{2\pi} \sigma}e^{-\frac{s^2}{2}}~d(s\sigma+\mu)\\
&=-\frac{1}{\alpha}\int_{-\infty}^{z_{\alpha}}
\frac{s\sigma+\mu}{\sqrt{2\pi}}e^{-\frac{s^2}{2}}~ds\\
&=-\frac{\sigma}{\alpha}\int_{-\infty}^{z_{\alpha}}
\frac{s}{\sqrt{2\pi}}e^{-\frac{s^2}{2}}~ds-\frac{\mu}{\alpha}\int_{-\infty}^{z_{\alpha}}
\frac{1}{\sqrt{2\pi}}e^{-\frac{s^2}{2}}~ds\\
&=-\frac{\sigma}{\alpha}\phi(z_{\alpha})-\frac{\mu}{\alpha}\alpha\\
&=-\frac{\sigma}{\alpha}\phi(z_{\alpha})-\mu\\
ES(\alpha)&=\frac{\sigma\phi(z_{\alpha})}{\alpha}+\mu\\
\end{split}

***(b)
 Suppose that the yearly return on a stock is normally distributed with mean 0.04 and standard deviation 0.18. If one purchases $100,000 worth of this stock, what is ES(0.05) with T equal to one year?***

the **loss** on this stock is normally distributed with mean **-0.04** and standard deviation 0.18<br>
$z_{0.05}=-1.64485$ and $\phi(z_{0.05})=0.1031356$.<br><br>
\begin{split}
ES(\alpha)&= 100000\times[\mu + \sigma\phi(z_{\alpha})/\alpha]\\
&=100000\times[-0.04 + 0.18\times\phi(z_{0.05})/0.05]\\
&=100000\times[(-0.04 + 0.18\times0.1031356/0.05]\\
&=100000\times0.3312883\\
&=33128.83\\
\end{split}

***(c)
Suppose that the yearly returns on stocks A and B are both normally distributed with mean 0.04 and standard deviation 0.18. Assume further that they are jointly normal with correlation 0.2. Consider a portfolio of investing \$50,000 in A and \$50,000 in B, what is ES(0.05) with T equal to one year? Compare the result with (b).***

the **loss** on stocks A and B are both normally distributed with mean **-0.04**.<br>
$z_{0.05}=-1.64485$ and $\phi(z_{0.05})=0.1031356$.<br>
$w_A=w_B=0.5$.<br><br>
\begin{split}
\mu_{A+B}
&=-0.04\times0.5-0.04\times0.5=-0.04\\
\sigma_{A+B}&=\sqrt{(w_A\sigma_A)^2+(w_
B\sigma_B)^2+2\rho w_A\sigma_A w_B\sigma_B}\\
&=\sqrt{(0.5\times0.18)^2+(0.5\times0.18)^2+
2\times0.2\times0.5\times0.18\times0.5\times0.18}\\
&=0.1394274\\\\
ES(\alpha)&= (50000+50000)\times[\mu + \sigma\phi(z_{\alpha})/\alpha]\\
&=100000\times[-0.04 + 0.1394274\times\phi(z_{0.05})/0.05]\\
&=100000\times[-0.04 + 0.1394274\times0.1031356/0.05]\\
&=100000\times0.24759857\\
&=24759.857\\
\end{split}
The result is much smaller than ES(0.05) in (b), i.e the expected shortfall is reduced which means risks has been reduced.